### PR TITLE
[rubysrc2cpg] Added ast support for keyword parameters and test-cases

### PR DESCRIPTION
**Pull request by karan-batavia**
@xavierpinho, tagging for review.

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 1]
```diff
@@ -1120,3 +1124,5 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
     callNode.columnNumber shouldBe Some(9)
   }
 }
+
+class Test extends RubyCode2CpgFixture {}
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1128]: Should be removed

**Review by xavierpinho — CHANGES_REQUESTED** [review]

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 2]
```diff
@@ -1039,22 +1039,26 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
...
+  "have correct structure when RHS for a mandatory parameter is provided" in {
     val cpg = code("""
         |def foo(bar: world)
         |end
         |""".stripMargin)
 
-    cpg.method("foo").parameter.size shouldBe 1
+    val List(parameterNode) = cpg.method("foo").parameter.l
+    parameterNode.name shouldBe "bar"
+    parameterNode.lineNumber shouldBe Some(2)
```
[the remark this answers is not in the document]
Comment (karan-batavia) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1061]: We're collecting only the names for all kinds of parameters atm. We'll have to change the modelling for each kind of parameter if we make this change. Can I create a separate PR for that?

[the remark this answers is not in the document]
Comment (xavierpinho) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1061]: Sure, just please leave a comment or create an issue, whatever you prefer, so this doesn't get lost.

[the remark this answers is not in the document]
Comment (karan-batavia) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1062]: Done. @xavierpinho. Please merge this.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 3]
```diff
@@ -1056,8 +1056,9 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
         |end
         |""".stripMargin)
 
+    println(cpg.parameter.code.l)
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala]: Remove this println please

**Review by xavierpinho — APPROVED** [review]

**Pull request merged**